### PR TITLE
feat(api-key-service): create new app

### PR DESCRIPTION
**Pull request by jakim929**
creates new trpc api service for the api-key-service

mostly uses the trpc-api generator but changes up how we do dependency injection for metrics / logger since we no longer have to do hacky "setLoggingServer"

if changes look good i can change the generator files as well

### CODE REVIEW

**Comment by jakim929** [comment]
* **#224** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/ecosystem/224?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @jakim929 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/api-key-service/src/routes/Route.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+   * The handler specifies the typescript types for input and output
+   * and writes the function to fetch the data
+   */
+  public abstract readonly handler:
+    | BuildProcedure<any, any, any>
+    | CreateRouterInner<any, any>
+  constructor(
+    protected readonly trpc: Trpc,
+    protected readonly logger: Logger,
+    protected readonly metrics: Metrics,
```
Comment (jakim929) [inline apps/api-key-service/src/routes/Route.ts:37]: main difference from the current generator - injects loggers / metrics without need for setLoggingServer

File: apps/api-key-service/src/constants/session.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import { type IronSession, type SessionOptions } from 'iron-session'
+
+import { envVars } from './envVars'
+
+export const sessionOptions: SessionOptions = {
+  password: envVars.IRON_SESSION_SECRET,
+  cookieName: `api-key-service.${envVars.DEPLOYMENT_ENV}.auth.session-token`,
```
Comment (nitaliano) [inline apps/api-key-service/src/constants/session.ts:7]: we can probably remove all the cookie related logic, if we need anything on the admin api side of things we'll just proxy it thru the dapp console api to this service

**Review by nitaliano — APPROVED** [review]

File: apps/api-key-service/src/constants/session.ts (hunk 2 above)
Comment (jakim929) [reply apps/api-key-service/src/constants/session.ts]: updated to remove all admin api related code

**Pull request merged**